examples/ex-07.c

```diff
@@ -0,0 +1,772 @@
+      }
+   }
+}
+
+/*-------------------------*
+ | my integration routines |
+ *-------------------------*/
+
+// returns the derivative at point *u*
+void Lorenz(const VEC u, VEC u_out)
+{
+   u_out[0] = sigma * (u[1] - u[0]);
+   u_out[1] = u[0] * (rho - u[2]) - u[1];
+   u_out[2] = u[0] * u[1] - beta * u[2];
+}
+
+// returns the Jacobian at point *u*
+void Lorenz_du(const VEC u, MAT J)
+{
+   J[0][0] = -sigma;
+   J[0][1] = sigma;
+   J[0][2] = 0.;
+   J[1][0] = rho - u[2];
+   J[1][1] = -1.;
+   J[1][2] = -u[0];
+   J[2][0] = u[1];
+   J[2][1] = u[0];
+   J[2][2] = -beta;
+}
+
+/*
+ * Computes a time-step with euler's method
+ * Optionally computes the jacobian of the time-step and
+ * writes the result in F (if not NULL)
+ */
+void Euler(VEC u0, double h, MAT *F)
+{
+   VEC k1;
+   Lorenz(u0, k1);
+
+   if (F) // compute Jacobian of the time-step
+   {
+      MAT J;
+      MatSetEye(*F);
+
+      // F = I + hJ(u0)
+      Lorenz_du(u0, J);
+      MatAxpy(h, J, *F);
+   }
+
+   // u1 = u0 + hf(u0)
+   VecAxpy(h, k1, u0);
+}
+
+/*--------------------------------------------------------------------------
+ * User-defined routines and structures required by Braid
+ *--------------------------------------------------------------------------*/
+
+/* can put anything in my app and name it anything as well */
+typedef struct _braid_App_struct
+{
+   MPI_Comm comm;
+   double tstart;
+   double tstop;
+   int ntime;
+   int rank;
+   double lyap_exps[VecSize]; // temporary storage for lyapunov exponents
+   FILE *file;                // saves the state vectors
+   FILE *file_lv;             // saves the Lyapunov vectors and exponents
+
+} my_App;
+
+int my_App_Init(my_App *app,
+                MPI_Comm comm_,
+                double tstart_,
+                double tstop_,
+                int ntime_,
+                int rank_,
+                FILE *file,
+                FILE *file_lv)
+{
+   (app->comm) = comm_;
+   (app->tstart) = tstart_;
+   (app->tstop) = tstop_;
+   (app->ntime) = ntime_;
+   (app->rank) = rank_;
+   (app->file) = file;
+   (app->file_lv) = file_lv;
+   for (int i = 0; i < rank_; i++)
+   {
+      (app->lyap_exps)[i] = 0.;
+   }
+
+   return 0;
+}
+
+/* can put anything in my vector and name it anything as well */
+typedef struct _braid_Vector_struct
+{
+   VEC values;
+} my_Vector;
+
+int my_Step(braid_App app,
+            braid_Vector ustop,
+            braid_Vector fstop,
+            braid_Vector u,
+            braid_StepStatus status)
+{
+   double tstart; /* current time */
+   double tstop;  /* evolve to this time */
+   braid_StepStatusGetTstartTstop(status, &tstart, &tstop);
+
+   double h; /* dt value */
+   h = tstop - tstart;
+
+   // get the number of Lyapunov vectors we need to propagate
+   int rank; /* rank of Delta correction */
+   braid_StepStatusGetDeltaRank(status, &rank);
+   MAT LinProp = {{0., 0., 0.}, {0., 0., 0.}, {0., 0., 0.}};
+
+   if (rank > 0) // we are propagating Lyapunov vectors
```

Comment:
I would give a comment somewhere specifically spelling out what the user has to do in this function, e.g., something like

For delta correction, the user must propagate (i) the solution vector (as in traditional Braid) and (ii) the Lyapunov vectors.  The Lyapunov vectors are accessible through the status structure and are advanced with the Jacobian (see below)

Also, is the comment "we are propagating Lyapunov vectors" in the right spot?  It seems to come right before you propagate the solution vector.